### DISCUSSION TITLE
Use ZMQ simulation to implement the multi-UE functionality of GNB

### DISCUSSION THREAD

**Discussion by small-hulu** [srsran/srsRAN_Project#311] [General Help]
Hello, I'm very interested in multi-UE testing, can I use ZMQ to make multiple UEs attach to gnb, and if not, is there any other way? I've looked at other discussions, and in the SA networking of srsran, it is possible to use GNUradio to implement multiple UE, but can this pattern be applied to gnb？

**Comment by pgawlowicz** [comment]
@small-hulu Please check the new section of our gnb+srsUE tutorial that describes how to emulate multiple-UEs:
https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation

Please remember to clone the newest version of the srsRAN-4G repository.
